apps/api/v2/src/ee/overlay-calendars/services/overlay-calendars.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Calendar } from "@/ee/overlay-calendars/inputs/get-overlay-calendars-busy-times.input";
+import { CredentialsRepository } from "@/modules/credentials/credentials.repository";
+import { Injectable, InternalServerErrorException, UnauthorizedException } from "@nestjs/common";
+import { User } from "@prisma/client";
+import { DateTime } from "luxon";
+
+import { getBusyCalendarTimes } from "@calcom/platform-libraries";
+import { ValueType } from "@calcom/platform-types";
+
+@Injectable()
+export class OverlayCalendarsService {
```

Comment:
Should the service be called `CalendarsService`? Functions like `getUniqCalendarCredentials` and `getCalendarsWithCredentials` sounds generic and applicable to all calendars (my guess).

Also, maybe then we should not have `ee/overlay-calendars` folder but just `ee/calendars` and also simply `/calendars` endpoint that returns data, and then is it used in overlay calendar or not depends on consumer if possible.